msal_extensions/cache_lock.py

```diff
@@ -12,6 +13,8 @@ class CrossPlatLock(object):
     """
     def __init__(self, lockfile_path):
         self._lockpath = lockfile_path
+        # Support for passing through arguments to the open syscall was added in v1.4.0
+        open_kwargs = {'buffering': 0} if inspect.signature(portalocker.Lock).parameters.get('file_open_kwargs') else {}
```

Comment:
You could just use `portalocker.__version__` for the version check. Not bullet-proof, but good enough. And it won't break our Python 2 support.

Reply:
Ok, done - I was trying to be clever, but there's really no need